src/PageBanner.jsx

```diff
@@ -1,197 +1,136 @@
-/**
-@class PageBanner
-to set shim, make sure that <PageBanner /> is at the top of your page.
-*/
 import React from 'react';
 import classnames from 'classnames';
-import './waypoints';
+import { List } from 'immutable';
+
+import BabyBanner from './BabyBanner';
 
-let waypoint;
+import './waypoints';
 
 export default React.createClass({
-  getDefaultProps: function() {
+  displayName: 'PageBanner',
+  waypoint: null,
+
+  propTypes: {
+    pageMessages: React.PropTypes.object,
+    triggerClose: React.PropTypes.number,
+    triggerOpen: React.PropTypes.number
+  },
+
+  getDefaultProps() {
     return {
-      message: '',
-      type: 'success',
-      duration: 3000,
-      afterClose: () => {},
-      topOffset: null,
-      topPalmOffset: null,
-      hideShim: false,
-      sticky: false,
-      closeIconClass: '',
-      tabIndexCloseIcon: () => '-1',
-      ariaLabelCloseIcon: 'Close Icon',
-      roleCloseIcon: 'button',
-      onKeyUpCloseIcon: () => {},
-      ariaLiveMessage: 'off',
-      ariaHidden: true,
-      roleMessage: null,
+      pageMessages: List(),
       triggerClose: 0,
       triggerOpen: 0,
-      tabIndexBody: '-1'
+      hideShim: false
     };
   },
 
-  propTypes: {
-    message: React.PropTypes.any,
-    type: React.PropTypes.string,
-    duration: React.PropTypes.number,
-    afterClose: React.PropTypes.func,
-    topOffset: React.PropTypes.string,
-    topPalmOffset: React.PropTypes.string,
-    hideShim: React.PropTypes.bool,
-    sticky: React.PropTypes.bool,
-    closeIconClass: React.PropTypes.string,
-    tabIndexCloseIcon: React.PropTypes.func,
-    ariaLabelCloseIcon: React.PropTypes.any,
-    roleCloseIcon: React.PropTypes.string,
-    onKeyUpCloseIcon: React.PropTypes.func,
-    ariaLiveMessage: React.PropTypes.string,
-    ariaHidden: React.PropTypes.bool,
-    roleMessage: React.PropTypes.string,
-    triggerClose: React.PropTypes.number,
-    triggerOpen: React.PropTypes.number,
-    tabIndexBody: React.PropTypes.string
-  },
-
   getInitialState() {
     return {
-      closePageBannerTimer: null,
       isShowing: false,
       isFixed: false,
-      height: null,
-      tabIndexCloseIcon: '-1',
-      ariaHidden: this.props.ariaHidden
+      ariaHidden: true
     };
   },
 
-  componentDidMount() {
-    //set the height of the banner to animate in and out correctly
-    const el = this.refs.pageBannerBody;
-    var height = el.clientHeight;
-    el.style.top = `${-(height)}px`;
+  componentWillReceiveProps(nextProps) {
+    const { triggerClose, triggerOpen } = this.props;
 
-    this.setState({ height });
+    if (triggerOpen !== nextProps.triggerOpen) {
+      this.setState({
+        isShowing: true,
+        ariaHidden: false
+      });
+    }
 
-    waypoint = new Waypoint({
-      element: this.refs.pageBanner,
-      handler: function(direction) {
-        this._handleWaypoint(direction);
-      }.bind(this)
-    });
+    if (triggerClose !== nextProps.triggerClose) {
```

Comment:
i think this should be an elseif...it doesn't make sense that another dev would triggerOpen then triggerClose